src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt

```diff
@@ -33,6 +33,12 @@ interface RsExpandedElement : RsElement {
             }
             return parent
         }
+
+        fun getContextImplWithoutIndexAccess(psi: RsExpandedElement): PsiElement? {
```

Comment:
Maybe it's better to add `isIndexAccessForbidden: Boolean = false` to `getContextImpl` and avoid code duplication?